### PR TITLE
resolve: fix potential dead lock

### PR DESCRIPTION
**Pull request by BusyJay**
If a store is failed to resolved, it should be removed from resolving set.

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @overvenus

File: src/server/server.rs [hunk 1]
```diff
@@ -228,6 +233,13 @@ mod tests {
         }
     }
 
+    pub fn is_unreachable_to(msg: &SignificantMsg, region_id: u64, to_peer_id: u64) -> bool {
```
Comment (overvenus) [inline src/server/server.rs]: Revoke `pub`.

**Comment by disksing** [comment]
LGTM.

**Comment by siddontang** [comment]
CI failed @BusyJay

**Comment by BusyJay** [comment]
Not related to the changes.

**Pull request merged**